### PR TITLE
Again support WDAG in Microsoft Edge.

### LINKED ISSUE #15164: Edge in Defender Application Guard not accessible to NVDA at all
https://github.com/nvaccess/nvda/pull/7600 suggests that NVDA should work with Defender Application Guard (WDAG/MDAG) windows, however the current version produces no speech output when hovering elements or using NVDA-A to say all. As detailed below, I have narrowed this down to it working in 2019.2.1 and being broken from 2019.3 onwards.
### Steps to reproduce:
1. Enable Application Guard through "Turn Windows features on or off", selecting "Microsoft Defender Application Guard", going through the necessary reboots, etc.
2. Start Edge
3. Access the "Settings and More" menu (ALT-F), "New Application Guard window"
4. In the newly appeared Edge window, navigate to http://nvaccess.org and attempt to read the page
### Actual behavior:

NVDA remains silent.
### Expected behavior:

Expect to hear page contents.
### NVDA logs, crash dumps and other attachments:
[nvda.log](https://github.com/nvaccess/nvda/files/12088304/nvda.log) from 2023.1, enabled debug and UIA logging.
#### NVDA installed/portable/running from source:
Tested both installed and running from the installer.
#### NVDA version:
2023.1.0.27913
#### Windows version:
Windows 10 22H2 19045.3208,
Windows 11 22H2 22621.1992
#### Name and version of other software in use when reproducing the issue:
Microsoft Edge 114.0.1823.86
#### Other information about your system:
These are two separate machines running different versions of Windows.
#### Does the issue still occur after restarting your computer?
Yes
#### Have you tried any other versions of NVDA? If so, please report their behaviors.
Yes. The behaviour works with 2019.2.1 and breaks with 2019.3 onwards.
#### If NVDA add-ons are disabled, is your problem still occurring?
No add-ons are enabled.
#### Does the issue still occur after you run the COM Registration Fixing Tool in NVDA's tools menu?
Did not attempt this.

### PR DESCRIPTION
**Pull request by michaelDCurran**
### Link to issue number:
Fixes #15164

### Summary of the issue:
NVDA's support for Windows Defender Application Guard (WDAG) no longer workds as the requirement that the remote WDAG UIA tree be parented into the local tree under the main WDAG window is lon longer true. Thus it is impossible to find the local WDAG window and use this window handle for all WDAG NVDAObjects.
This means that NVA is essentially  silent while focus is within a WDAG applcation.
 
### Description of user facing changes
NVDA can again interact with applications such as Microsoft edge when run in Windows Defender Application Guard.

### Description of development approach
* When trying to find the window handle for a WDAG UIA element, rather than walking up the ancestry, instead use the currently active window, if it is a WDAG host window and its for the same instance as the UIA element.
* Improve and refactor the part of UIA NVDAObject's findOverlayClasses method that deals with Chrome objects, to match on a UIA framework of 'Chrome', not just of a particular Chrome window class. Code needed to be also reordered so that more specific cases such as documents and lists where looked for first, before the more generic Chrome UIA classes.

### Testing strategy:
* Opened Microsoft Edge.
* Opened a New Defender Application Guard window from the edge Settings button.
* Navigated to https://www.nvaccess.org/
* Ensured that NVDA reported focus while tabbing around the page.

### Known issues with pull request:
* Browse mode is not available much of the time, due to IUIAutomationTextPattern::rangeFromChild failing on the document's text pattern when given an element within the page. This bug can be also reproduced with Narrator with Edge and WDAG. When outside of WDAG, this bug is not seen, thus it looks like some kind of marshalling bug for WDAG.
* As we can only locate the correct WDAG window if it is the active window, we will ignore UIA events from background WDAG windows. Generally we want this anyway, but it is worth noting.

### Code Review Checklist:

- [x] Documentation:
  - Change log entry
  - User Documentation
  - Developer / Technical Documentation
  - Context sensitive help for GUI changes
- [x] Testing:
  - Unit tests
  - System (end to end) tests
  - Manual testing
- [x] UX of all users considered:
  - Speech 
  - Braille
  - Low Vision
  - Different web browsers
  - Localization in other languages / culture than English
- [x] API is compatible with existing add-ons.
- [x] Security precautions taken.

### CODE REVIEW

**Comment by leonardder** [comment]
Will this still work in the old case, i.e. when the remote WDAG UIA tree is parented into the local tree?

**Comment by michaelDCurran** [comment]
@leonardder Assuming the user is interacting with the WDAG process in the foreground (I.e. focus is inside it) then yes. Parenting has broken multiple times over WDAG's history in Windows. Thus this is a much safer, though slightly more limited option.

**Comment by michaelDCurran** [comment]
It may be some time before the rangeFromChild bug is fixed in Windows. But merging this beforehand still provides some benefits for accessing WDAG anyway.

File: source/UIAHandler/__init__.py [hunk 1]
```diff
@@ -77,6 +77,8 @@
 
 # The name of the WDAG (Windows Defender Application Guard) process
 WDAG_PROCESS_NAME=u'hvsirdpclient'
+# The window class  of the WDAG (Windows Defender Application Guard) main window
+WDAG_WINDOW_CLASS_NAME = u'RAIL_WINDOW'
```
Comment (seanbudd) [inline source/UIAHandler/__init__.py:81]: Unicode marking is unnecessary in python 3

```suggestion
# The window class of the WDAG (Windows Defender Application Guard) main window
WDAG_WINDOW_CLASS_NAME = "RAIL_WINDOW"
```

**Review by seanbudd — APPROVED** [review]

**Pull request merged**